compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -756,41 +756,114 @@ object Parsers {
      *   4. there is at least one token between the braces
      *   5. the closing brace is also at the end of the line, or it is followed by one of
      *      `then`, `else`, `do`, `catch`, `finally`, `yield`, or `match`.
-     *   6. the opening brace does not follow a `=>`. The reason for this condition is that
-     *      rewriting back to braces does not work after `=>` (since in most cases braces are omitted
-     *      after a `=>` it would be annoying if braces were inserted).
-     */
-    def bracesToIndented[T](body: => T, rewriteWithColon: Boolean): T = {
+     *   6. the opening brace does not follow a closing brace
+     *   7. last token is not a leading operator
+     *   8. not a block in a sequence of statements
+     *   9. cannot rewrite if colon required after a NEWLINE, e.g.
+     *     true ||
+     *     {
+     *       false
+     *     }
+     */
+    def bracesToIndented[T](body: => T, inStatSeq: Boolean, rewriteWithColon: Boolean): T =
+      import IndentRewriteState.*
+      val prevSaved = prev.saveCopy
```

Comment:
This is where we need the last token. So that we can patch it later, if it's an operator that will be followed by a :.

For instance, object *:  { is rewritten to object `*:`:.